presentation/src/main/java/com/knocklock/presentation/lockscreen/LockScreenNotificationListener.kt

```diff
@@ -0,0 +1,56 @@
+package com.knocklock.presentation.lockscreen
+
+import android.app.Notification
+import android.service.notification.NotificationListenerService
+import android.service.notification.StatusBarNotification
+import android.text.TextUtils
+import com.knocklock.data.repository.NotificationRepositoryImpl
+import com.knocklock.domain.repository.NotificationRepository
+import kotlinx.coroutines.*
+import com.knocklock.domain.model.Notification as NotificationDomainModel
+
+/**
+ * @Created by 김현국 2022/12/04
+ * @Time 5:50 PM
+ */
+
+class LockScreenNotificationListener : NotificationListenerService() {
+
+    private val job by lazy { SupervisorJob() }
+    private val scope by lazy { CoroutineScope(Dispatchers.IO + job) }
+    private lateinit var notificationRepository: NotificationRepository
+    override fun onCreate() {
+        super.onCreate()
+        notificationRepository = NotificationRepositoryImpl(this) // data layer 의존성 제거 hilt로
```

Comment:
```suggestion
       //TODO: data layer 의존성 제거 hilt로
        notificationRepository = NotificationRepositoryImpl(this)
```

TODO 주석으로 해도 좋을 것 같아요. 이렇게 해두면 IDE에서 TODO 주석만 찾을 수 있어요!

Reply:
@jaeryo2357 주석 반영할께용